diff --git a/index.js b/index.js
index aadfc6c1715c76e2dbfaf35a51ff711cbe92fcdf..7e2ee1583daffdfcf1763ddc399b7a57f0a5339d 100644
--- a/index.js
+++ b/index.js
@@ -130,6 +130,9 @@ const create = (win, options) => {
 				visible: properties.mediaType === 'image',
 				click(menuItem) {
 					properties.srcURL = menuItem.transform ? menuItem.transform(properties.srcURL) : properties.srcURL;
+					if(properties.srcURL.startsWith("https://media.discordapp.net")){
+						properties.srcURL = /^.*(?=(&=&))/.exec(properties.srcURL)
+					}
 					download(win, properties.srcURL);
 				},
 			}),
